app/src/components/Editor.tsx

```diff
@@ -150,49 +177,123 @@ function Document() {
   );
 }
 
-const ParagraphContainer = styled.div``;
+const DocumentTitle = styled.h1`
+  text-align: left;
+  font-weight: normal;
+  font-size: 20px;
+  grid-column-start: 2;
+`;
+function FileNameDisplay({ path }: { path: string }) {
+  const extension = extname(path);
+  const base = basename(path, extension);
+
+  return (
+    <DocumentTitle>
+      {base}
+      <span style={{ fontWeight: 'lighter' }}>.{extension}</span>
+    </DocumentTitle>
+  );
+}
+
+const ParagraphContainer = styled.div`
+  user-select: none;
+`;
 const SpeakerContainer = styled.div`
-  overflow-wrap: break-word;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`;
+const LongSilenceSpan = styled.span`
+  padding: 0 8px;
+  color: transparent;
+  background: center / auto 80% no-repeat url(${quarterRest});
 `;
-function Silence(): JSX.Element {
+function LongSilence(props: { selected: boolean } & HTMLAttributes<HTMLSpanElement>): JSX.Element {
+  console.log(quarterRest);
+  return (
+    <LongSilenceSpan
+      className={'item'}
+      style={props.selected ? { backgroundColor: 'lightblue' } : {}}
+      {...props}
+    >
+      {' '}
+    </LongSilenceSpan>
+  );
+}
+function ShortSilence(props: { selected: boolean } & HTMLAttributes<HTMLSpanElement>): JSX.Element {
   return (
-    <img
-      className={'word'}
-      style={{ height: '1em', filter: 'var(--filter)' }}
-      src={quarterRest}
-      alt={'quarter rest'}
-    />
+    <span
+      className={'item'}
+      style={props.selected ? { backgroundColor: 'lightblue' } : {}}
+      {...props}
+    >
+      {' '}
+    </span>
   );
 }
+const Word = styled.span<{ selected: boolean }>`
+  ${(props) =>
+    props.selected &&
+    css`
+      background-color: lightblue;
```

Comment:
Here (and in the other elements), we need to differentiate between light and dark mode. In dark mode, this produces white text on lightblue background, which is hard to read
<img width="278" src="https://user-images.githubusercontent.com/6545097/133343047-016b7b91-b9ad-407a-8792-b78df40f267d.png">